test/multisigwallet.test.ts

```diff
@@ -0,0 +1,519 @@
+        transactionId
+      );
+      assert.isTrue(new BigNumber(0).eq(confirmationCount));
+    });
+
+    it('should revert for non-owner', async () => {
+      await assertReverts(async () => {
+        await wallet.revokeConfirmation(1, { from: accounts[9] });
+      });
+    });
+
+    it('should revert for non-existing transaction', async () => {
+      await assertReverts(async () => {
+        await wallet.revokeConfirmation(2, { from: owners[0] });
+      });
+    });
+
+    it('should revert for not confirmed transaction', async () => {
+      await assertReverts(async () => {
+        await wallet.revokeConfirmation(2, { from: owners[0] });
+      });
+    });
+  });
+
+  describe('#executeTransaction', () => {
+    let transactionId: AnyNumber;
+
+    beforeEach(async () => {
+      const tx = await submitAddOwner({ newOwner: accounts[9] });
+      transactionId = findLastTransactionId(tx);
+      await wallet.confirmTransaction(transactionId, { from: owners[1] });
+    });
+
+    it('should emit ExecutionEvent', async () => {
+      const tx = await wallet.confirmTransaction(transactionId, {
+        from: owners[2]
+      });
+
+      const log = findLastLog(tx, 'Execution');
+      assert.isOk(log);
+
+      const event = log.args as ExecutionEvent;
+      assert.isOk(event);
+    });
+
+    it('should set transaction as executed', async () => {
+      await wallet.confirmTransaction(transactionId, {
+        from: owners[2]
+      });
+
+      const transactionCount = await wallet.getTransactionCount(false, true);
+      assert.isTrue(new BigNumber(1).eq(transactionCount));
+
+      const transactionIds = await wallet.getTransactionIds(0, 1, false, true);
+      assertNumberEqual(transactionId, transactionIds[0]);
+    });
+
+    it('should revert for non-owner', async () => {
+      await assertReverts(async () => {
+        await wallet.executeTransaction(transactionId, { from: accounts[9] });
+      });
+    });
+
+    it('should revert for non-existing transaction', async () => {
+      await assertReverts(async () => {
+        await wallet.executeTransaction(2, { from: owners[0] });
+      });
+    });
+
+    it('should revert for not confirmed transaction', async () => {
+      await assertReverts(async () => {
+        await wallet.executeTransaction(transactionId, { from: owners[2] });
+      });
+    });
+  });
+
+  describe('#changeRequirement', () => {
+    const newRequired = new BigNumber(2);
+
+    it('should emit ExecutionFailureEvent for invalid requirements', async () => {
+      let result: ExecutionResult;
+      let log: TransactionLog;
+      let event: ExecutionFailureEvent;
+      result = await executeChangeRequirements(new BigNumber(0));
+      log = findLastLog(result.lastTransaction, 'ExecutionFailure');
+      assert.isOk(log);
+
+      event = log.args;
+      assertNumberEqual(result.transactionId, event.transactionId);
+
+      result = await executeChangeRequirements(new BigNumber(4));
```

Comment:
A test case should test only a single thing. You test 0 and 4 requirements in one tests.